exit/test_exit.c

```diff
@@ -0,0 +1,665 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    POSIX.1-2017 standard library functions tests
+ *    HEADER:
+ *    - stdlib.h
+ *    - unistd.h
+ *    TESTED:
+ *    - exit()
+ *    - _exit() (_Exit equivalent)
+ *    - atexit()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <pthread.h>
+
+#include <unity_fixture.h>
+
+#define TEST_EXIT_PATH1 "/tmp/exit_test_file1"
+#define TEST_EXIT_PATH2 "/tmp/exit_test_file2"
+#define TEST_EXIT_PATH3 "/tmp/exit_test_file3"
+#define TEST_EXIT_PATH4 "/tmp/exit_test_file4"
+#define TEST_EXIT_PATH5 "/tmp/exit_test_file5"
+#define TEST_EXIT_STR   "test123"
+
+/*
+	Aspects required by POSIX, which wasn't tested:
+		- all open named semaphores in the calling process shall be closed
+		  -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/806
+
+		- threads terminated by a call to _Exit() or _exit() shall not invoke their cancellation cleanup handlers
+		  -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/827
+
+		- if the exit of the process causes a process group to become orphaned, and if any member of the newly-orphaned
+		  process group is stopped, then a SIGHUP signal followed by a SIGCONT signal shall be sent to each process in
+		  the newly-orphaned process group -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/809
+ */
+
```

Comment:
And the rest of points - I haven't verified the rest yet.
![Screenshot from 2023-09-18 18-32-06](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/73d995a2-5af1-4682-aa7d-6614c87e87fc)